sections/main-search.liquid

```diff
@@ -44,6 +44,14 @@
   assign sort_by = search.sort_by | default: search.default_sort_by
   assign terms = search.terms | escape
   assign search_url = '?q=' | append: terms | append: '&options%5Bprefix%5D=last&sort_by=' | append: sort_by
+  
+  if settings.spacing_grid_horizontal == 0 and settings.spacing_grid_vertical == 0 and settings.card_style == 'card'
+    assign collapsed_class = 'grid--collapsed grid--collapsed-x grid--collapsed-y'
+  elsif settings.spacing_grid_horizontal == 0
+    assign collapsed_class = 'grid--collapsed grid--collapsed-x'
+  elsif settings.spacing_grid_vertical == 0 and settings.card_style == 'card'
+    assign collapsed_class = 'grid--collapsed grid--collapsed-y'
+  endif
```

Comment:
I don't want it in theme.liquid because of the possibility of local overrides in the future. I also didn't make it a snippet originally because of the number of exceptions between collage and blog specific names and the standard card exception. With the latest changes, I've gone ahead and put them into a snippet. However blog and collage still have the classes inline.